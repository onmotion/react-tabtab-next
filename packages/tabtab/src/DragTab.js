"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_sortable_hoc_1 = require("react-sortable-hoc");
var Tab_1 = __importDefault(require("./Tab"));
var DragTabElement = (0, react_sortable_hoc_1.SortableElement)(function (_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(Tab_1.default, __assign({}, props, { index: props.tabIndex }), children));
});
var DragTab = (function (_super) {
    __extends(DragTab, _super);
    function DragTab() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DragTab.prototype.render = function () {
        var _this = this;
        var _a = this.props, children = _a.children, props = __rest(_a, ["children"]);
        return (React.createElement(DragTabElement, __assign({ index: 0, ref: function (node) { return _this.__DRAG_TAB_INTERNAL_NODE = node; } }, props), children));
    };
    return DragTab;
}(React.PureComponent));
exports.default = DragTab;
