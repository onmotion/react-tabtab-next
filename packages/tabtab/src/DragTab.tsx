// @flow
import React, { FC, forwardRef, memo } from 'react';
import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';

interface Props {
    id: string;
    activeIndex?: number;
    index?: number;
    children: React.ReactNode;
}
const DragTab: FC<Props> = memo(
    forwardRef<HTMLElement, Props>(({ children, id, index, activeIndex, ...rest }, ref) => {
        const { attributes, listeners, setNodeRef, transform, transition } = useSortable({ id: id });

        const style = {
            transform: CSS.Transform.toString(transform),
            transition,
            // cursor: 'default',
        };

        return (
            <div ref={setNodeRef} style={style} {...attributes} {...listeners}>
                {React.cloneElement(children as React.ReactElement, {
                    ...rest,
                    key: id,
                    active: index === activeIndex,
                    index,
                    tabIndex: id,
                    ref: ref,
                })}
            </div>
        );
    })
);

export default DragTab;
