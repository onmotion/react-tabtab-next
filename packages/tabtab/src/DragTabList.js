"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var SortMethod_1 = __importDefault(require("./SortMethod"));
var react_sortable_hoc_1 = require("react-sortable-hoc");
var TabList_1 = __importDefault(require("./TabList"));
var DragTabContainer = (0, react_sortable_hoc_1.SortableContainer)(function (_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (react_1.default.createElement(TabList_1.default, __assign({}, props), children));
});
var DragTabList = (function (_super) {
    __extends(DragTabList, _super);
    function DragTabList() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DragTabList.prototype.render = function () {
        var _a = this.props, children = _a.children, props = __rest(_a, ["children"]);
        return (react_1.default.createElement(DragTabContainer, __assign({ onSortEnd: this.onSortEnd, axis: 'x', lockAxis: 'x', pressDelay: 100 }, props), children));
    };
    return DragTabList;
}(SortMethod_1.default));
exports.default = DragTabList;
DragTabList.displayName = 'DragTabList';
