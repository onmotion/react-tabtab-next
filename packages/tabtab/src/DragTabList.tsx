// @ts-nocheck
import React from "react"
import SortMethod from "./SortMethod"
import { SortableContainer } from "react-sortable-hoc"
import TabList from "./TabList"

const DragTabContainer = SortableContainer(({ children, ...props }) => {
  return <TabList {...props}>{children}</TabList>
})

export default class DragTabList extends SortMethod {
  render() {
    const { children, ...props } = this.props
    return (
      <DragTabContainer
        onSortEnd={this.onSortEnd}
        axis="x"
        lockAxis="x"
        // if no pressDelay, close button cannot be triggered,
        // because it would always treat click as dnd action
        pressDelay={100}
        {...props}
      >
        {children}
      </DragTabContainer>
    )
  }
}

DragTabList.displayName = "DragTabList"
