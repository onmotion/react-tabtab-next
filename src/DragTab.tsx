// @flow
import * as React from 'react';
import {SortableElement} from 'react-sortable-hoc';
import Tab from './Tab';

const DragTabElement = SortableElement(({children, ...props}) => {
  return (
    // @ts-ignore
    <Tab index={props.tabIndex} {...props}>
      {children}
    </Tab>
  )
});



class DragTab extends React.PureComponent {

  __DRAG_TAB_INTERNAL_NODE: React.ElementRef<any>;

  render() {
    const {children, ...props} = this.props;
    return (
      <DragTabElement ref={node => this.__DRAG_TAB_INTERNAL_NODE = node} {...props}>
        {children}
      </DragTabElement>
    )
  }
}

// DragTab.displayName = 'DragTab';

export default DragTab;