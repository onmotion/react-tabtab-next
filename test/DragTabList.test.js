import React from 'react';
import {mount} from 'enzyme';
import DragTabList from '../src/DragTabList';
import DragTab from '../src/DragTab';
import tabListTest from './tabListTest';
import toJson from 'enzyme-to-json';

describe('DragTabList', () => {
  tabListTest('DragTab', DragTabList, DragTab);

  test('DragTabList should have required props', () => {
    const component = mount(
      <DragTabList customStyle={{}}>
        <DragTab>Tab1</DragTab>
      </DragTabList>
    );
    expect(typeof toJson(component).node.rendered.props.pressDelay).toBe('number');
  })
});

