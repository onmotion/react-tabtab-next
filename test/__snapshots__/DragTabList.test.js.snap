// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragTabList custom ListStyle style 1`] = `
.c0 {
  background-color: white;
  text-align: left;
  position: relative;
  white-space: nowrap;
  overflow: hidden;
  width: auto;
  padding: 0 0px 0 0px;
}

.c1 {
  overflow: hidden;
}

.c2 {
  padding-left: 0;
  position: relative;
  margin: 0;
  list-style: none;
  display: inline-block;
  -webkit-transition: -webkit-transform .3s cubic-bezier(.42,0,.58,1);
  -webkit-transition: transform .3s cubic-bezier(.42,0,.58,1);
  transition: transform .3s cubic-bezier(.42,0,.58,1);
}

.c3 {
  display: inline-block;
  padding: 10px 15px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  cursor: pointer;
  color: black;
}

.c4 {
  vertical-align: middle;
}

<DragTabList
  customStyle={
    Object {
      "CustomTabList": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": false,
          "rules": Array [
            "
      background-color: white;
      text-align: left;
      position: relative;
      white-space: nowrap;
      overflow: hidden;
      width: auto;
      padding: ",
            [Function],
            ";
    ",
            "
          background-color: red;
        ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    }
  }
>
  <sortableList
    axis="x"
    customStyle={
      Object {
        "CustomTabList": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": false,
            "rules": Array [
              "
        background-color: white;
        text-align: left;
        position: relative;
        white-space: nowrap;
        overflow: hidden;
        width: auto;
        padding: ",
              [Function],
              ";
      ",
              "
            background-color: red;
          ",
            ],
          },
          "displayName": "Styled(styled.div)",
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      }
    }
    distance={0}
    getHelperDimensions={[Function]}
    hideSortableGhost={true}
    lockAxis="x"
    lockOffset="50%"
    lockToContainerEdges={false}
    onSortEnd={[Function]}
    pressDelay={100}
    pressThreshold={5}
    shouldCancelStart={[Function]}
    transitionDuration={300}
    useWindowAsScrollContainer={false}
  >
    <Component
      customStyle={
        Object {
          "CustomTabList": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gZMcBi",
              "isStatic": false,
              "rules": Array [
                "
          background-color: white;
          text-align: left;
          position: relative;
          white-space: nowrap;
          overflow: hidden;
          width: auto;
          padding: ",
                [Function],
                ";
        ",
                "
              background-color: red;
            ",
              ],
            },
            "displayName": "Styled(styled.div)",
            "render": [Function],
            "styledComponentId": "sc-gZMcBi",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        }
      }
    >
      <TabList
        customStyle={
          Object {
            "CustomTabList": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-gZMcBi",
                "isStatic": false,
                "rules": Array [
                  "
            background-color: white;
            text-align: left;
            position: relative;
            white-space: nowrap;
            overflow: hidden;
            width: auto;
            padding: ",
                  [Function],
                  ";
          ",
                  "
                background-color: red;
              ",
                ],
              },
              "displayName": "Styled(styled.div)",
              "render": [Function],
              "styledComponentId": "sc-gZMcBi",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          }
        }
      >
        <div>
          <styled.div
            hasExtraButton={false}
          >
            <StyledComponent
              forwardedClass={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c0",
                    "rules": Array [
                      "
                  background-color: white;
                  text-align: left;
                  position: relative;
                  white-space: nowrap;
                  overflow: hidden;
                  width: auto;
                  padding: ",
                      [Function],
                      ";
                ",
                    ],
                  },
                  "displayName": "styled.div",
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              hasExtraButton={false}
            >
              <div
                className="c0"
              >
                <styled.div>
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": true,
                          "lastClassName": "c1",
                          "rules": Array [
                            "
                        overflow: hidden;
                      ",
                          ],
                        },
                        "displayName": "styled.div",
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={[Function]}
                  >
                    <div
                      className="c1"
                    >
                      <styled.ul
                        role="tablist"
                      >
                        <StyledComponent
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "c2",
                                "rules": Array [
                                  "
                              padding-left: 0;
                              position: relative;
                              margin: 0;
                              list-style: none;
                              display: inline-block;
                              transition: transform .3s cubic-bezier(.42, 0, .58, 1);
                            ",
                                ],
                              },
                              "displayName": "styled.ul",
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "ul",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={[Function]}
                          role="tablist"
                        >
                          <ul
                            className="c2"
                            role="tablist"
                          >
                            <DragTab
                              active={false}
                              index={0}
                              key="0/.0"
                              tabIndex={0}
                            >
                              <sortableElement
                                active={false}
                                collection={0}
                                index={0}
                                tabIndex={0}
                              >
                                <Component
                                  active={false}
                                  tabIndex={0}
                                >
                                  <Tab
                                    active={false}
                                    index={0}
                                    tabIndex={0}
                                  >
                                    <styled.li
                                      active={false}
                                      aria-controls="react-tabtab-panel-0"
                                      aria-selected={false}
                                      id="react-tabtab-tab-0"
                                      onClick={[Function]}
                                      role="tab"
                                    >
                                      <StyledComponent
                                        active={false}
                                        aria-controls="react-tabtab-panel-0"
                                        aria-selected={false}
                                        forwardedClass={
                                          Object {
                                            "$$typeof": Symbol(react.forward_ref),
                                            "attrs": undefined,
                                            "componentStyle": ComponentStyle {
                                              "componentId": "sc-EHOje",
                                              "isStatic": false,
                                              "lastClassName": "c3",
                                              "rules": Array [
                                                "
                                            display: ",
                                                [Function],
                                                ";
                                            ",
                                                [Function],
                                                "
                                          
                                            user-select: none;
                                            &:hover {
                                              cursor: pointer;
                                              color: black;
                                            }
                                          ",
                                              ],
                                            },
                                            "displayName": "styled.li",
                                            "render": [Function],
                                            "styledComponentId": "sc-EHOje",
                                            "target": "li",
                                            "toString": [Function],
                                            "warnTooManyClasses": [Function],
                                            "withComponent": [Function],
                                          }
                                        }
                                        forwardedRef={[Function]}
                                        id="react-tabtab-tab-0"
                                        onClick={[Function]}
                                        role="tab"
                                      >
                                        <li
                                          aria-controls="react-tabtab-panel-0"
                                          aria-selected={false}
                                          className="c3"
                                          id="react-tabtab-tab-0"
                                          onClick={[Function]}
                                          role="tab"
                                        >
                                          <styled.span>
                                            <StyledComponent
                                              forwardedClass={
                                                Object {
                                                  "$$typeof": Symbol(react.forward_ref),
                                                  "attrs": undefined,
                                                  "componentStyle": ComponentStyle {
                                                    "componentId": "sc-bZQynM",
                                                    "isStatic": true,
                                                    "lastClassName": "c4",
                                                    "rules": Array [
                                                      "
                                                  vertical-align: middle;
                                                ",
                                                    ],
                                                  },
                                                  "displayName": "styled.span",
                                                  "render": [Function],
                                                  "styledComponentId": "sc-bZQynM",
                                                  "target": "span",
                                                  "toString": [Function],
                                                  "warnTooManyClasses": [Function],
                                                  "withComponent": [Function],
                                                }
                                              }
                                              forwardedRef={null}
                                            >
                                              <span
                                                className="c4"
                                              >
                                                Tab1
                                              </span>
                                            </StyledComponent>
                                          </styled.span>
                                        </li>
                                      </StyledComponent>
                                    </styled.li>
                                  </Tab>
                                </Component>
                              </sortableElement>
                            </DragTab>
                          </ul>
                        </StyledComponent>
                      </styled.ul>
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </TabList>
    </Component>
  </sortableList>
</DragTabList>
`;

exports[`DragTabList render DragTab List 1`] = `
.c0 {
  background-color: white;
  text-align: left;
  position: relative;
  white-space: nowrap;
  overflow: hidden;
  width: auto;
  padding: 0 0px 0 0px;
}

.c1 {
  overflow: hidden;
}

.c2 {
  padding-left: 0;
  position: relative;
  margin: 0;
  list-style: none;
  display: inline-block;
  -webkit-transition: -webkit-transform .3s cubic-bezier(.42,0,.58,1);
  -webkit-transition: transform .3s cubic-bezier(.42,0,.58,1);
  transition: transform .3s cubic-bezier(.42,0,.58,1);
}

.c3 {
  display: inline-block;
  padding: 10px 15px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  cursor: pointer;
  color: black;
}

.c4 {
  vertical-align: middle;
}

<DragTabList
  customStyle={Object {}}
>
  <sortableList
    axis="x"
    customStyle={Object {}}
    distance={0}
    getHelperDimensions={[Function]}
    hideSortableGhost={true}
    lockAxis="x"
    lockOffset="50%"
    lockToContainerEdges={false}
    onSortEnd={[Function]}
    pressDelay={100}
    pressThreshold={5}
    shouldCancelStart={[Function]}
    transitionDuration={300}
    useWindowAsScrollContainer={false}
  >
    <Component
      customStyle={Object {}}
    >
      <TabList
        customStyle={Object {}}
      >
        <div>
          <styled.div
            hasExtraButton={false}
          >
            <StyledComponent
              forwardedClass={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c0",
                    "rules": Array [
                      "
                  background-color: white;
                  text-align: left;
                  position: relative;
                  white-space: nowrap;
                  overflow: hidden;
                  width: auto;
                  padding: ",
                      [Function],
                      ";
                ",
                    ],
                  },
                  "displayName": "styled.div",
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              hasExtraButton={false}
            >
              <div
                className="c0"
              >
                <styled.div>
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": true,
                          "lastClassName": "c1",
                          "rules": Array [
                            "
                        overflow: hidden;
                      ",
                          ],
                        },
                        "displayName": "styled.div",
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={[Function]}
                  >
                    <div
                      className="c1"
                    >
                      <styled.ul
                        role="tablist"
                      >
                        <StyledComponent
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "c2",
                                "rules": Array [
                                  "
                              padding-left: 0;
                              position: relative;
                              margin: 0;
                              list-style: none;
                              display: inline-block;
                              transition: transform .3s cubic-bezier(.42, 0, .58, 1);
                            ",
                                ],
                              },
                              "displayName": "styled.ul",
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "ul",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={[Function]}
                          role="tablist"
                        >
                          <ul
                            className="c2"
                            role="tablist"
                          >
                            <DragTab
                              active={false}
                              index={0}
                              key="0/.0"
                              tabIndex={0}
                            >
                              <sortableElement
                                active={false}
                                collection={0}
                                index={0}
                                tabIndex={0}
                              >
                                <Component
                                  active={false}
                                  tabIndex={0}
                                >
                                  <Tab
                                    active={false}
                                    index={0}
                                    tabIndex={0}
                                  >
                                    <styled.li
                                      active={false}
                                      aria-controls="react-tabtab-panel-0"
                                      aria-selected={false}
                                      id="react-tabtab-tab-0"
                                      onClick={[Function]}
                                      role="tab"
                                    >
                                      <StyledComponent
                                        active={false}
                                        aria-controls="react-tabtab-panel-0"
                                        aria-selected={false}
                                        forwardedClass={
                                          Object {
                                            "$$typeof": Symbol(react.forward_ref),
                                            "attrs": undefined,
                                            "componentStyle": ComponentStyle {
                                              "componentId": "sc-EHOje",
                                              "isStatic": false,
                                              "lastClassName": "c3",
                                              "rules": Array [
                                                "
                                            display: ",
                                                [Function],
                                                ";
                                            ",
                                                [Function],
                                                "
                                          
                                            user-select: none;
                                            &:hover {
                                              cursor: pointer;
                                              color: black;
                                            }
                                          ",
                                              ],
                                            },
                                            "displayName": "styled.li",
                                            "render": [Function],
                                            "styledComponentId": "sc-EHOje",
                                            "target": "li",
                                            "toString": [Function],
                                            "warnTooManyClasses": [Function],
                                            "withComponent": [Function],
                                          }
                                        }
                                        forwardedRef={[Function]}
                                        id="react-tabtab-tab-0"
                                        onClick={[Function]}
                                        role="tab"
                                      >
                                        <li
                                          aria-controls="react-tabtab-panel-0"
                                          aria-selected={false}
                                          className="c3"
                                          id="react-tabtab-tab-0"
                                          onClick={[Function]}
                                          role="tab"
                                        >
                                          <styled.span>
                                            <StyledComponent
                                              forwardedClass={
                                                Object {
                                                  "$$typeof": Symbol(react.forward_ref),
                                                  "attrs": undefined,
                                                  "componentStyle": ComponentStyle {
                                                    "componentId": "sc-bZQynM",
                                                    "isStatic": true,
                                                    "lastClassName": "c4",
                                                    "rules": Array [
                                                      "
                                                  vertical-align: middle;
                                                ",
                                                    ],
                                                  },
                                                  "displayName": "styled.span",
                                                  "render": [Function],
                                                  "styledComponentId": "sc-bZQynM",
                                                  "target": "span",
                                                  "toString": [Function],
                                                  "warnTooManyClasses": [Function],
                                                  "withComponent": [Function],
                                                }
                                              }
                                              forwardedRef={null}
                                            >
                                              <span
                                                className="c4"
                                              >
                                                Tab1
                                              </span>
                                            </StyledComponent>
                                          </styled.span>
                                        </li>
                                      </StyledComponent>
                                    </styled.li>
                                  </Tab>
                                </Component>
                              </sortableElement>
                            </DragTab>
                          </ul>
                        </StyledComponent>
                      </styled.ul>
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </TabList>
    </Component>
  </sortableList>
</DragTabList>
`;
